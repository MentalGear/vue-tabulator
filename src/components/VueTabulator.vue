<template>
  <div
    v-if="tabulatorInstance"
    ref="table"
    class="tabulator"
  />
</template>


<script lang='ts'>
import {
  Component, Prop, Vue, Watch, Model,
} from 'vue-property-decorator';
import Tabulator from 'tabulator-tables';

@Component({
  name: 'VueTabulator',
})
export default class VueTabulator extends Vue {
  @Model('change', { default: () => [] })
  public tableData?: Array<any>;

  private tabulatorInstance: any = {};

  @Prop({ default: () => ({}) })
  private options: any;

  private resolvedOptions: any = {};

  public getInstance() {
    return this.tabulatorInstance;
  }

  @Watch('tableData', { deep: true })
  private createTable() {
    this.tabulatorInstance = new Tabulator(
      this.$refs.table,
      this.resolvedOptions,
    );
  }

  @Watch('options', { deep: true })
  private updateOptions() {
    this.resolvedOptions = {
      ...this.options,
      data: this.tableData,
    };

    this.createTable();
  }



  mounted() {
    this.updateOptions();
  }
}
</script>


<style lang='scss'>
</style>
