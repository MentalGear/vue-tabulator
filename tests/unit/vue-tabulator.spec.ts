import { shallowMount } from '@vue/test-utils';
import VueTabulator from '@/components/VueTabulator.vue';
import Vue from 'vue';

describe('VueTabulator.vue', () => {
  it('renders correct based in the options', () => {
    const options = {
      columns: [
        {
          title: 'Name',
          field: 'name',
          sorter: 'string',
          width: 200,
          editor: true,
        },
      ],
    };

    const wrapper = shallowMount(VueTabulator, {
      propsData: {
        options,
      },
    });


    expect(wrapper.findAll('.tabulator-col-title').filter(item => item.text().match('Name')).length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
