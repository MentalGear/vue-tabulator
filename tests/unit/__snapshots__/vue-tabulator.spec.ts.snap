// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VueTabulator.vue renders correct based in the options 1`] = `
<div class="tabulator" role="grid" tabulator-layout="fitData">
  <div class="tabulator-header">
    <div class="tabulator-headers">
      <div class="tabulator-col tabulator-sortable" role="columnheader" aria-sort="none" tabulator-field="name" style="min-width: 40px; width: 200px;" title="">
        <div class="tabulator-col-content">
          <div class="tabulator-col-title">Name</div>
          <div class="tabulator-arrow"></div>
        </div>
        <div class="tabulator-col-resize-handle"></div>
        <div class="tabulator-col-resize-handle prev"></div>
      </div>
    </div>
    <div class="tabulator-frozen-rows-holder"></div>
  </div>
  <div class="tabulator-tableHolder" tabindex="0">
    <div class="tabulator-table"></div>
  </div>
</div>
`;
